changedetectionio/html_tools.py

```diff
@@ -67,10 +67,35 @@ def element_removal(selectors: List[str], html_content):
 
 
 # Return str Utf-8 of matched rules
-def xpath_filter(xpath_filter, html_content, append_pretty_line_formatting=False):
+def xpath_filter(xpath_filter, html_content, is_xml=False, append_pretty_line_formatting=False):
     from lxml import etree, html
+    # is_xml is a user flag.
+
+    # This is default case for html_requests.
+    if type(html_content) is bytes and not is_xml:
+        tree = etree.HTML(html_content)
+    # There are special XML cases that need a user intervention(is_xml).
+    elif type(html_content) is bytes and is_xml:
+        tree = etree.XML(html_content)
+    # If type is str.
+    else:
+        # XML(uses XML parser) or XHTML(uses HTML parser) may have XML encoding declaration.
+        # So remove the encoding declaration only.
+        re_xml_encoding_declaration = re.compile(r'(?<=\<\?xml version)( *= *([\'\"])[^\1\ \n]*?\2 )encoding *= ?([\'\"])[^\1\n]*?\2 *')
+        html_content = re_xml_encoding_declaration.sub('\g<1>', html_content)
```

Comment:
I will append a test for this regex pattern. However, I want you to suggest kinds of tests.
I already have patterns I tested.

Reply:
Also, I noticed the `\g<1>` has a deprecation warning. I will fix it later.